http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcTest.java

```diff
@@ -146,24 +150,63 @@ public void testSucessfulAuthenticationWithEnvironmentVariableExpression() throw
+    public void testInvalidScope() throws Exception {
+        String expectedScope = OIDC_SCOPE + "+INVALID_SCOPE";
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET, "INVALID_SCOPE"), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), "error=invalid_scope", expectedScope, true);
+    }
+
+    @Test
+    public void testEmptyScope() throws Exception {
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET, ""), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT, OIDC_SCOPE, false);
+    }
+
+    @Test
+    public void testSingleScopeValue() throws Exception {
+        String expectedScope = OIDC_SCOPE + "+profile";
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET, "profile"), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT, expectedScope, false);
     }
 
+    @Test
+    public void testMultipleScopeValue() throws Exception {
+        String expectedScope = OIDC_SCOPE + "+phone+profile+email";
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET, "email phone profile"), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT, expectedScope, false);
+    }
+
+    @Test
+    public void testOpenIDScopeValue() throws Exception {
+        String expectedScope = OIDC_SCOPE;
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET, OIDC_SCOPE), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT, expectedScope, false);
+    }
+
+    @Test
+    public void testOpenIDWithMultipleScopeValue() throws Exception {
+        String expectedScope = OIDC_SCOPE + "+phone+profile+email";//order gets changed when combining with query parameters
+        performAuthentication(getOidcConfigurationInputStreamWithScope(CLIENT_SECRET,  "email phone profile " + OIDC_SCOPE), KeycloakConfiguration.ALICE, KeycloakConfiguration.ALICE_PASSWORD,
+                true, HttpStatus.SC_MOVED_TEMPORARILY, getClientUrl(), CLIENT_PAGE_TEXT, expectedScope, false);
+    }
     private void performAuthentication(InputStream oidcConfig, String username, String password, boolean loginToKeycloak,
-                                       int expectedDispatcherStatusCode, String expectedLocation, String clientPageText) throws Exception {
+                                       int expectedDispatcherStatusCode, String expectedLocation, String clientPageText, String expectedScope, boolean checkInvalidScopeError) throws Exception {
```

Comment:
We can add a new performAuthentication method with the old parameters that just calls this performAuthentication with all of the passed in parameters and with expectedScope and checkInvalidScopeError set to null and false.

That way, you won't need to update all of the existing calls to performAuthentication above.